extensions/blocks/podcast-player/components/with-error-boundary.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+export default function withErrorBoundary( WrappedComponent ) {
+	class ErrorBoundary extends Component {
+		state = {
+			didError: false,
+		};
+
+		componentDidCatch = ( error, errorInfo ) => {
+			this.props.onError( error, errorInfo );
+		};
+
+		static getDerivedStateFromError = () => {
+			return { didError: true };
+		};
+
+		render() {
+			if ( this.state.didError ) {
+				return (
+					<section className="jetpack-podcast-player">
+						<p className="jetpack-podcast-player__error">
+							{ __(
+								'An unexpected error occured within the Podcast Player. Please reload the page.',
```

Comment:
Not sure about the `Please reload the page.` part. In the unlikely case that the user would see this message, can we be sure that a reload doesn't cause the same error? In that case, seeing the same message again might be frustrating to the user. Maybe we could change the wording slightly to suggest a reload _might_ fix the issue.

Reply:
sounds good!